dashboard/test/lib/pd/jot_form/translation_test.rb

```diff
@@ -112,6 +112,57 @@ class TranslationTest < ActiveSupport::TestCase
         assert_equal 'This is the summary for scale1', questions.first.text
       end
 
+      test 'get_questions uses question name as fallback when no text is available' do
+        JotFormRestClient.any_instance.expects(:get_questions).with(@form_id).returns(
+          {
+            content: {
+              1 => {
+                qid: 1,
+                type: 'control_scale',
+                name: 'intentionallyLeftBlank',
+                text: '',
+                order: 1
+              }
+            }
+          }.deep_stringify_keys
+        )
+
+        questions = Translation.new(@form_id).get_questions
+        assert_equal 1, questions.length
+        assert_equal 1, questions.first.id
+        assert_equal 'intentionallyLeftBlank', questions.first.name
+        assert_equal 'intentionallyLeftBlank', questions.first.text
+      end
+
+      test 'get_questions prefers -summary text to fallback when no text is available' do
+        JotFormRestClient.any_instance.expects(:get_questions).with(@form_id).returns(
+          {
+            content: {
+              1 => {
+                qid: 1,
+                type: 'control_scale',
+                name: 'intentionallyLeftBlank',
+                text: '',
+                order: 1
+              },
+              2 => {
+                qid: 2,
+                type: 'control_text',
+                name: 'intentionallyLeftBlank-summary',
+                text: 'A real description',
+                order: 2
+              }
+            }
+          }.deep_stringify_keys
+        )
+
+        questions = Translation.new(@form_id).get_questions
+        assert_equal 1, questions.length
```

Comment:
Total ignorance here: are there not two questions listed above (with `qid` 1 & 2)?

Reply:
When we parse JotForm "questions" we ignore certain question types...

https://github.com/code-dot-org/code-dot-org/blob/a4f0fcb35797321d8f012851f3f0eb837286d84d/dashboard/lib/pd/jot_form/translation.rb#L96-L97

...including type `control_text` which is simply a label, not a question (although it _can_ be the source for another question's text, as tested here).  If you're curious, here's the complete set of ignored question types:

https://github.com/code-dot-org/code-dot-org/blob/3f077edf563c9686ef0153771fa11f6c6760b25c/dashboard/lib/pd/jot_form/constants.rb#L15-L23